python/MDSplus/tests/data_case.py

```diff
@@ -135,10 +133,6 @@ def _executeTests(test, a, b, res, **kwargs):
                 test('cosd(_a)',    m.COSD(a),   res.pop(), **kwargs)
                 test('tand(_a)',    m.TAND(a),   res.pop(), **kwargs)
                 test('anint(_a/3)', m.ANINT(a/3), res.pop(), **kwargs)
-            else:  # strip results of real operations
-                res = res[:-rcount]
-        else:  # strip results of floating operations
-            res = res[:-acount]
         if real:  # some binary operations are only defined for real numbers
```

Comment:
This `real` flag and the associated comment are also confusing.   Note that "real" numbers is the entire number line from negative infinity to positive infinity (i.e., on a computer, that is all floating point numbers and integers).   Modulo operations are typically only done on the "integers".  (However, in Python one can indeed apply `mod` to floating point numbers.)

Reply:
Yes, the `real` flag is always true except for the tests that use complex number types, for which the operations wouldn't be defined.

Reply:
Also makes sense.   Thanks for the answer.